pkg/virt-handler/vm.go

```diff
@@ -2721,6 +2722,22 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 		}
 	}
 
+	isolationRes, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return fmt.Errorf(failedDetectIsolationFmt, err)
+	}
+
+	// Backwards compatibility
+	//TODO: remove this block of code when we do not support updates from old versions.
+	vmPath := filepath.Join("/proc", strconv.Itoa(isolationRes.Pid()), "/root")
+	oldDownwardMetricDisk := filepath.Join(config.DownwardAPIDisksDir, config.VhostmdName)
+	_, err = os.Stat(filepath.Join(vmPath, oldDownwardMetricDisk))
+
+	if !pkgerrors.Is(err, os.ErrNotExist) {
+		// Updating from old version. Let's restore the old path
+		config.DownwardMetricDisk = oldDownwardMetricDisk
```

Comment:
You can't do this as this is a global variable and we are running multiple goroutines.

Reply:
You are right. I have changed the approach to something more thread safe I believe.